extensions/amp-powr-player/0.1/amp-powr-player.js

```diff
@@ -0,0 +1,432 @@
+      // We still need to check this.iframe_ as the component may have
+      // been unlaid out by now.
+      if (this.iframe_ && this.iframe_.contentWindow) {
+        this.iframe_.contentWindow. /*OK*/ postMessage(JSON.stringify(dict({
+          'command': command,
+          'args': arg,
+        })), 'https://player.powr.com');
+      }
+    });
+  }
+
+  /**
+   * @param {!Event} event
+   * @private
+   */
+  handlePlayerMessage_(event) {
+    const {element} = this;
+
+    if (event.source != this.iframe_.contentWindow) {
+      return;
+    }
+
+    const eventData = getData(event);
+    if (!isJsonOrObj(eventData)) {
+      return;
+    }
+
+    const data = objOrParseJson(eventData);
+    if (data === undefined) {
+      return; // We only process valid JSON.
+    }
+
+    const eventType = data['event'];
+    if (!eventType) {
+      return;
+    }
+
+    if (eventType === 'ready') {
+      this.onReady_(data);
+    }
+
+    if (eventType === 'playing') {
+      this.playing_ = true;
+    }
+    if (eventType === 'pause') {
+      this.playing_ = false;
+    }
+
+    if (redispatch(element, eventType, {
+      'ready': VideoEvents.LOAD,
+      'playing': VideoEvents.PLAYING,
+      'pause': VideoEvents.PAUSE,
+      'ended': VideoEvents.ENDED,
+      'ads-ad-started': VideoEvents.AD_START,
+      'ads-ad-ended': VideoEvents.AD_END,
+    })) {
+      return;
+    }
+
+    if (eventType === 'volumechange') {
+      const muted = data['muted'];
+      if (muted == null ||
+          this.muted_ == muted) {
+        return;
+      }
+      this.muted_ = muted;
+      element.dispatchCustomEvent(mutedOrUnmutedEvent(this.muted_));
+      return;
+    }
+  }
+
+  /**
+   * @param {!JsonObject} data
+   * @private
+   */
+  onReady_(data) {
+    this.hasAmpSupport_ = true;
+
+    Services.timerFor(this.win)
+        .cancel(this.readyTimeout_);
+
+    const {element} = this;
+
+    installVideoManagerForDoc(element);
+    Services.videoManagerForDoc(element).register(this);
+
+    dev().info(TAG,
+        'Player %s ready. ' +
+        'Powr Player version: %s IFrame Support version: %s',
+        this.playerId_, data['powrVersion'], data['iframeVersion']
+    );
+  }
+
+  /**
+   * @return {string}
+   * @private
+   */
+  getIframeSrc_() {
+    const {element: el} = this;
+    const account = user().assert(
+        el.getAttribute('data-account'),
+        'The data-account attribute is required for <amp-powr-player> %s',
+        el);
+
+    this.playerId_ = user().assert(
+        el.getAttribute('data-player'),
+        'The data-player attribute is required for <amp-powr-player> %s',
+        el);
+
+    const video = el.getAttribute('data-video');
+    const terms = el.getAttribute('data-terms');
+
+    user().assert(
+        video || terms,
+        'The data-video or data-terms attribute is required for ' +
+        '<amp-powr-player> %s',
+        el);
+
+    const src = 'https://player.powr.com/iframe.html?' +
```

Comment:
Nit: use `addParamsToUrl`:

```js
const srcPrefix = 'https://player.powr.com/iframe.html';

const srcParams = dict({
  'account': account,
  'player': this.playerId_,
});

if (video) {
   srcParams['video'] = video;
}
if (terms) {
   srcParams['terms'] = terms;
}

const src = addParamsToUrl(srcPrefix, srcParams);
```

Reply:
Solid idea. Makes it a lot easier to follow.